Modules/termios.c

```diff
@@ -315,6 +315,96 @@ termios_tcflow_impl(PyObject *module, int fd, int action)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+termios.tcgetwinsize
+
+    fd: fildes
+    /
+
+Get the tty winsize for file descriptor fd.
+
+Returns a list [ws_row, ws_col].
+[clinic start generated code]*/
+
+static PyObject *
+termios_tcgetwinsize_impl(PyObject *module, int fd)
+/*[clinic end generated code: output=31825977d5325fb6 input=c7ed8aa957d108c0]*/
+{
+#if defined(TIOCGWINSZ)
```

Comment:
You could also use TIOCGSIZE if that exists like https://github.com/NetBSD/src/blob/trunk/lib/libedit/terminal.c#L927

Reply:
Thank you for this. I have made the requested changes.

Reply:
It looks like there is also a `TIOCSSIZE`. I will update `termios.tcsetwinsize()` to take that into consideration as well. Note: https://fossies.org/linux/dialog/ttysize.c makes their code look compact by using some #define directives before their function definition. However, I decided to avoid copying their style.

Reply:
Done.